presto-main/src/main/java/io/prestosql/server/GenerateTraceTokenRequestFilter.java

```diff
@@ -29,22 +29,28 @@
 public class GenerateTraceTokenRequestFilter
         implements HttpRequestFilter
 {
+    private static final int SEQUENCE_NUMBER_HEX_LENGTH = 10;
+
     private final String prefix = randomUUID().toString().toLowerCase(ENGLISH).replace("-", "");
     private final AtomicLong sequence = new AtomicLong();
 
     @Override
     public Request filterRequest(Request request)
     {
         requireNonNull(request, "request is null");
-        String newToken = createToken(sequence.getAndIncrement());
         return fromRequest(request)
-                .setHeader(TRACETOKEN_HEADER, newToken)
+                .setHeader(TRACETOKEN_HEADER, createToken(sequence.getAndIncrement()))
                 .build();
     }
 
     private String createToken(long value)
     {
-        return prefix + format("%010x", value);
+        StringBuilder builder = new StringBuilder(prefix.length() + SEQUENCE_NUMBER_HEX_LENGTH).append(prefix);
+        String sequenceNumHex = Long.toHexString(value);
+        for (int i = sequenceNumHex.length(); i < SEQUENCE_NUMBER_HEX_LENGTH; i++) {
+            builder.append('0');
```

Comment:
Padding on the right is not as good as padding on the left, eg 2nd and 11th numbers: 1, 10 will result in same token, right?

Reply:
The padding is on the left, since `sequenceNumHex` is appended below.